lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -120,6 +129,12 @@ StringRef FrontendInputsAndOutputs::getNameOfUniquePrimaryInputFile() const {
   return input == nullptr ? StringRef() : input->file();
 }
 
+std::string FrontendInputsAndOutputs::getStatsFileMangledInputName() const {
+  return isWholeModule()
+             ? "all"
+             : primaryInputCount() == 1 ? firstPrimaryInput().file() : "batch";
```

Comment:
Can you leave a FIXME here? Stats file names are supposed to be unique during a build, right?

Reply:
Done.